numpy/core/fromnumeric.py

```diff
@@ -293,7 +312,11 @@ def reshape(a, newshape, order='C'):
            [3, 4],
            [5, 6]])
     """
-    return _wrapfunc(a, 'reshape', newshape, order=order)
+    # Since it is hard to tell if `np.array` had to copy, the copy is only
+    # forced during reshape (but no-copy forced also for `np.array`).
+    return _wrapfunc_copy(
```

Comment:
I doubt introducing this `_wrapfunc_copy` is the nicest way to handle it, will think about it, but maybe someone has a nice idea.